azurerm/internal/services/cosmos/cosmosdb_sql_trigger_resource.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/cosmos/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceCosmosDbSQLTrigger() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCosmosDbSQLTriggerCreateUpdate,
+		Read:   resourceCosmosDbSQLTriggerRead,
+		Update: resourceCosmosDbSQLTriggerCreateUpdate,
+		Delete: resourceCosmosDbSQLTriggerDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SqlTriggerID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			"container_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SqlContainerID,
+			},
+
+			"body": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"operation": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(documentdb.All),
+					string(documentdb.Create),
+					string(documentdb.Update),
+					string(documentdb.Delete),
+					string(documentdb.Replace),
+				}, false),
+			},
+
+			"type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(documentdb.Pre),
+					string(documentdb.Post),
+				}, false),
+			},
+		},
+	}
+}
+func resourceCosmosDbSQLTriggerCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Cosmos.SqlResourceClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	containerId, _ := parse.SqlContainerID(d.Get("container_id").(string))
+	resourceGroup := containerId.ResourceGroup
+	accountName := containerId.DatabaseAccountName
+	containerName := containerId.ContainerName
+	databaseName := containerId.SqlDatabaseName
+	body := d.Get("body").(string)
+	triggerOperation := d.Get("operation").(string)
+	triggerType := d.Get("type").(string)
+
+	id := parse.NewSqlTriggerID(subscriptionId, resourceGroup, accountName, databaseName, containerName, name)
```

Comment:
we don't need to assign these to an intermediate?
```suggestion

	body := d.Get("body").(string)
	triggerOperation := d.Get("operation").(string)
	triggerType := d.Get("type").(string)

	id := parse.NewSqlTriggerID(subscriptionId, containerId.ResourceGroup, containerId.AccountName, containerId.DatabaseName, containerId.Name, name)
```